### PR TITLE
DPL-993 - Disallow deletion of libraries that are part of pools 

### PR DESCRIPTION
**Pull request by seenanair**
This PR includes
Added checks for  associated pools in a library before destroying a library

### CODE REVIEW

File: app/models/pacbio/library.rb [hunk 1]
```diff
@@ -101,5 +101,17 @@ def check_for_associated_wells
         throw(:abort)
       end
     end
+
+    def check_for_associated_pools
```
Comment (stevieing) [inline app/models/pacbio/library.rb:105]: Inwould think about turning this into a scope or a method in aliquot. That way everything is delegated to aliquot and you just need to return a boolean.

**Review by stevieing — CHANGES_REQUESTED** [review]
I think this would be better as a method in Aliquot. You could pass an argument which would be type in this case Pacbio::Pool and then it can be reused.

**Review by BenTopping — APPROVED** [review]
Nice one 👍 
We should hold off merging until the current work is through but hopefully that will be in the next few days.

File: app/models/pacbio/library.rb (hunk 1 above)
Comment (seenanair) [reply app/models/pacbio/library.rb:105]: 👍

**Review by stevieing — APPROVED** [review]

**Pull request merged**